### PR TITLE
[TCB] building the TCB with overflow checks

### PR DESCRIPTION
**Pull request by mimoo**
This proposes to build the TCB with [overflow checks](https://rust-lang.github.io/rfcs/1535-stable-overflow-checks.html).

How can I test this PR?

### CODE REVIEW

File: docker/build-common.sh [hunk 1]
```diff
@@ -14,17 +14,23 @@ export CARGO=$(rustup which --toolchain $RUST_NIGHTLY cargo)
...
          -p libra-operational-tool \
          -p libra-node \
-         -p libra-key-manager \
-         -p safety-rules \
          -p db-bootstrapper \
          -p backup-cli \
          "$@"
 
+# Build release TCB binaries
+git apply docker/tcb_build.patch
```
Comment (davidiw) [inline docker/build-common.sh:27]: what is tcb-build.patch?

Comment (mimoo) [reply docker/build-common.sh:27]: sorry, refresh!

Comment (mimoo) [reply docker/build-common.sh:27]: you're too fast!

**Comment by mimoo** [comment]
@davidiw looks like nothing is happening in longevity testing so far in terms of integer overflow, perhaps a good time to revisit this PR?

**Comment by davidiw** [comment]
Fix the linters, I think you need to either eliminate the last line in the patch. Let's verify all is good with a cluster test canary too. PR looks pretty interesting but good enough. Though might be good to get a dev-infra person to weigh in.

**Comment by mimoo** [comment]
It's ugly but without custom profiles (cf https://github.com/rust-lang/cargo/issues/6988) I don't see a better way to do it : o would be interested to know what the rust experts think : )

**Review by davidiw — APPROVED** [review]
god save us all

**Comment by davidiw** [comment]
let's not leave approved, stale PRs unclosed.

**Pull request closed without merging**